examples/basic/main.c

```diff
@@ -0,0 +1,1021 @@
+/*
+ * Copyright (c) The mldsa-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Import public mldsa-native API
+ *
+ * This requires specifying the parameter set and namespace prefix
+ * used for the build.
+ */
+#include "../../mldsa/common.h"
+#include "../../mldsa/sign.h"
+
+#include "test_only_rng/notrandombytes.h"
+
+#define CHECK(x)                                              \
+  do                                                          \
+  {                                                           \
+    int rc;                                                   \
+    rc = (x);                                                 \
+    if (!rc)                                                  \
+    {                                                         \
+      fprintf(stderr, "ERROR (%s,%d)\n", __FILE__, __LINE__); \
+      return 1;                                               \
+    }                                                         \
+  } while (0)
+
+int main(void)
+{
+  uint8_t pk[CRYPTO_PUBLICKEYBYTES];
+  uint8_t sk[CRYPTO_SECRETKEYBYTES];
+  uint8_t sig[CRYPTO_BYTES];
+  uint8_t sm[64 + CRYPTO_BYTES]; /* signed message buffer */
+  uint8_t m[64];                 /* message to sign */
+  uint8_t m2[64 + CRYPTO_BYTES]; /* recovered message buffer */
+  uint8_t ctx[16];               /* context string */
+  size_t siglen;
+  size_t smlen;
+  size_t mlen;
+  size_t ctxlen = 16;
+  const char *test_msg =
+      "This is a test message for ML-DSA digital signature algorithm!";
+  size_t msglen = strlen(test_msg);
+
+
+  /* The PCT modifies the PRNG state, so the KAT tests don't work.
+   * We run KAT tests only for disabled PCT. */
+#if !defined(MLD_CONFIG_KEYGEN_PCT)
+#if MLD_CONFIG_PARAMETER_SET == 44
+  const uint8_t expected_signature[] = {
```

Comment:
These long arrays are arther distracting given that the main purpose of the examples are to demonstrate how mldsa-native can be used? Should we move them to a separate file that is included? Or should we compare a hash of te expected signature.

Reply:
Good point, I did think the same, but thought it may be nice for users to see their full signatures etc. but I agree with you, and have just chosen to compare to the hash now